packages/constants/src/class-constants.php

```diff
@@ -72,7 +72,11 @@ public static function get_constant( $name ) {
 			return self::$set_constants[ $name ];
 		}
 
-		return defined( $name ) ? constant( $name ) : null;
+		if ( defined( $name ) ) {
+			return constant( $name );
+		}
+
+		return apply_filters( "jetpack_constant_$name", null, $name );
```

Comment:
Here is where I think we need to simplify things. The filter name doesn't have to depend on the `$name` because it's universal for every constant name. Let's call it something like `jetpack_constant_default_value`. This way we can assign the Utils method to that filter at load time and just forget about it. What do you think?

Also, this filter needs a docblock, please.

Reply:
I made the suggested changes in commit 0b39a42, including the filter docblock.  I agree that the code is much nicer now. 

However, I want to make sure that this filter is robust enough for future use with multiple plugins. Let’s say that another consuming plugin uses this filter to set some constant defaults, and that plugin’s filter function isn’t very careful about how it handles the input constant value. For example, it could just return `null` if it doesn’t have a default value for the constant.

If that plugin’s filter runs after `Utils::jetpack_api_constant_filter`, then the expected Jetpack API constant values wouldn’t be returned to the connection package. Instead, `null` would be returned.

What do you think? Is this a valid concern? Or should we expect that all filters will properly handle the constant values and won't overwrite the values returned by other filters?